### DISCUSSION TITLE
Any way to use CM with MVVM Toolkit?

### DISCUSSION THREAD

**Discussion by eeevans** [Caliburn-Micro/Caliburn.Micro#901] [Ideas]
I'm trying to use MVVM Toolkit for the INPC but can't figure a way to get it to work with IScreen and [ObservableProperty] or [ObservableObject].  I even considered forking MVVM Toolkit and removing rule to keep objects from already inheriting from INPC.

**Comment by vb2ae** [comment]
What is INPC?

**Reply by mvermef** [reply]
INotifyPropertyChanged

**Comment by eeevans** [comment]
They add INotifyPropertyChanged to the class and have a rule that errors if the class already inherits from the interface which IScreen does.

**Comment by vb2ae** [comment]
I dont think the mvvm community toolkit observable property attribute will work with caliburn micro.  i can create a Caliburn Micro extensions project and add a NotifyPropertyChanged attribute so it auto generates the notify property change code for you.  While I work on that you could look at Fody Property Changed.  (note I have not tried it with Caliburn Micro)

https://github.com/Fody/PropertyChanged

**Reply by eeevans** [reply]
Thanks so much for looking at it.  Don't go to the trouble of the extensions project.  That's what I was wresting with myself and wondering if there was something I wasn't seeing as a possible solution.  

I have used Fody previously but it adds to the build time for a large project and leads to non-obvious debug issues with the IL weaving.  We then moved to a container interceptor that would emit extended types with the INPC support at runtime with Proxy suffixed to the VM name and added a ViewLocator to strip it off when finding the associated View.  That was before the Roslyn source generators came out (.Net Framework 3.6 days) and affected the runtime performance significantly.  

I'll go back and revisit modifying the MVVM community toolkit and maybe offer a PR for consideration to them or just keep my own copy up to date for the purpose.  I'm hoping I can just short-circuit their check that the type doesn't already inherit from INotifyPropertyChanged and then modify the part that emits it to only add it if it doesn't already inherit from it in the chain.  I've not gone further than that though to find other possible issues.

Just FYI we did use Fody with CM and it worked fine for small number of ViewModels.

Again, much appreciated.
